maintenance/wikia/uncycloUsersMigrator.php

```diff
@@ -0,0 +1,695 @@
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->addOption( 'csv', 'Generate a report in CSV file' );
+		$this->addOption( 'dry-run', 'Don\'t perform any operations' );
+		$this->addOption( 'only-rename-global-users', 'Perform global users rename ONLY' );
+
+		$this->mDescription = 'This script migrates uncyclopedia user accounts to the shared database';
+	}
+
+	/**
+	 * Log helper
+	 *
+	 * @param string $msg
+	 * @param array $context
+	 */
+	private function info( $msg, Array $context = [] ) {
+		Wikia\Logger\WikiaLogger::instance()->info( $msg, $context );
+	}
+
+	/**
+	 * Error log helper
+	 *
+	 * @param string $msg
+	 * @param array $context
+	 */
+	protected function err( $msg, Array $context = [] ) {
+		Wikia\Logger\WikiaLogger::instance()->error( $msg, $context );
+	}
+
+	/**
+	 * Get the uncyclopedia database
+	 *
+	 * @param int $flag
+	 * @return DatabaseMysqli
+	 */
+	private function getUncycloDB($flag = DB_SLAVE) {
+		return wfGetDB($flag);
+	}
+
+	/**
+	 * Get the shared database
+	 *
+	 * @param int $flag
+	 * @return DatabaseMysqli
+	 */
+	private function getSharedDB($flag = DB_SLAVE) {
+		return wfGetDB( $flag, [], self::SHARED_DB );
+	}
+
+	/**
+	 * Get the global user object by the given user name
+	 *
+	 * @param string $userName
+	 * @return null|User
+	 */
+	private function getGlobalUserByName( $userName ) {
+		$row = $this->getSharedDB()->selectRow(
+			self::USER_TABLE,
+			'*',
+			[ 'user_name' => $userName],
+			__METHOD__
+		);
+
+		return $row ? User::newFromRow( (object) $row ) : null;
+	}
+
+	/**
+	 * Count all user edits performed after given date
+	 *
+	 * @param User $user user to count edits of
+	 * @param string $since MW timestamp string
+	 * @param int $since edits count
+	 */
+	protected function getEditsCountAfter( User $user, $since ) {
+		return $this->getUncycloDB()->selectField(
+			'revision',
+			'count(*)',
+			[
+				'rev_user' => $user->getId(),
+				sprintf( 'rev_timestamp > "%s"', $since )
+			],
+			__METHOD__
+		);
+	}
+
+	/**
+	 * Updates given column (userid_column or username_column) with a new value
+	 *
+	 * @param string $fieldType
+	 * @param string|int $oldValue
+	 * @param string|int $newValue
+	 */
+	protected function updateTables( $fieldType, $oldValue, $newValue ) {
+		$dbw = $this->getUncycloDB(DB_MASTER);
+
+		foreach( self::$mTableRenameRules as $entry ) {
+			// a single entry:
+			// array( 'table' => 'archive', 'userid_column' => 'ar_user', 'username_column' => 'ar_user_text' ),
+			if ( is_null( $entry[ $fieldType ] ) ) {
+				continue;
+			}
+
+			$columnName = $entry[ $fieldType ];
+			$tableName = $entry[ 'table' ];
+
+			if ( !$dbw->tableExists( $tableName ) ) {
+				$this->err( 'Table does not exist', [ 'tableName' => $tableName ] );
+				continue;
+			}
+
+			$dbw->update(
+				$entry[ 'table' ],
+				[ $columnName => $newValue ], // SET
+				[ $columnName => $oldValue ], // WHERE
+				__METHOD__
```

Comment:
Good point, I simply used the approach from the core RenameUser extension. Batches make more sense here.